src/CMakeLists.txt

```diff
@@ -36,6 +36,23 @@ add_dependencies(rayleigh run_param_header)
 
 target_include_directories(rayleigh PRIVATE ${PROJECT_SOURCE_DIR}/src/Include ${CMAKE_CURRENT_BINARY_DIR})
 
+set(RAYLEIGH_CPU_OPTIMIZATIONS "none" CACHE PATH "Specify a CPU architecture to build for. Currently only 'native' or 'none' are supported.")
+
+if (RAYLEIGH_CPU_OPTIMIZATIONS STREQUAL "native")
+  if(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
+    target_compile_options(rayleigh PRIVATE -march=native)
+    message(STATUS "enabling native optimizations")
+  elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
+    target_compile_options(rayleigh PRIVATE -xHost)
+  else()
+    message(FATAL_ERROR "For your compiler no native optimizations are automatically detected. Please specify compiler flags manually and set 'CPU_OPTIMIZATIONS' to 'none'")
+  endif()
+elseif (RAYLEIGH_CPU_OPTIMIZATIONS STREQUAL "none")
+  # do nothing, respect manually set compiler flags
+else()
+  message(FATAL_ERROR "Only 'native' and 'none' are supported for parameter 'CPU_OPTIMIZATIONS'")
+endif()
+
```

Comment:
Should we move this to the top-level `CMakeLists.txt` file?

Reply:
I cannot move it to the top-level file, because it needs to come after the `add_executable` file. I am not a fan of having too much in the `src/CMakeLists.txt` as well. What is the purpose of splitting the commands between the different files? Maybe we can merge the files, or at least move some of the things from `src/CMakeLists.txt` to `CMakeLists.txt`?

Reply:
Right. Let's leave it for now. We could probably even get away with a single `CMakeLists.txt` in Rayleigh's case. I would rearrange it later though.